### PR TITLE
fix: refactor language select

### PR DESCRIPTION
**Pull request by rbedeoan-plenty**
https://dev.azure.com/plentymarkets/plentymarkets/_sprints/backlog/BD%20-%20Team%20-%20LM/plentymarkets/Areas/Sprint%2043?workitem=80929

### CODE REVIEW

File: apps/web/lang/de.json [hunk 1]
```diff
@@ -171,7 +171,7 @@
   "quantitySelectorIncrease": "Menge erhöhen",
   "scrollTop": "Nach oben scrollen",
   "numberInCart": "{count} Produkte im Warenkorb",
-  "languageSelector": "Sprachauswahl",
+  "languageSelector": "Klicken Sie hier, um den modalen LanguageSelector zu öffnen",
```
Comment (FabianGerke) [inline apps/web/lang/de.json]: Why are we using a new translation here?

File: apps/web/lang/en.json [hunk 2]
```diff
@@ -171,7 +171,7 @@
   "quantitySelectorIncrease": "Increase value",
   "scrollTop": "Scroll to top",
   "numberInCart": "{count} product in cart",
-  "languageSelector": "Language selection",
+  "languageSelector": "Click to open languageSelector modal",
```
Comment (FabianGerke) [inline apps/web/lang/en.json]: Why are we using a new translation here?

File: apps/web/lang/en.json [hunk 3]
```diff
@@ -452,7 +452,7 @@
   },
   "softLogin": {
     "titleLogin": "Log in to your account to access your order",
-    "titlePostcode": "Please enter the postcode of an address stored for the order to view the order details.",
+    "titlePostcode": "Please enter the postcode of an address stored fore the order to view the order details.",
```
Comment (FabianGerke) [inline apps/web/lang/en.json]: ???

File: apps/web/components/LanguageSelector/LanguageSelector.vue [hunk 4]
```diff
@@ -1,39 +1,58 @@
...
+.mobileCountryFlags {
+  width: 10vw;
+}
+.desktopFlagLayout {
+  @apply hidden md:flex w-full bg-white items-center py-10 justify-center absolute z-[99999] top-[10%] shadow-xl;
+}
+.mobileFlagLayout {
+  @apply w-full bg-white flex items-center py-10 justify-center flex-col shadow-xl md:hidden;
+}
+</style>
```
Comment (FabianGerke) [inline apps/web/components/LanguageSelector/LanguageSelector.vue]: Why using css here? Use tailwind in the component itself.

File: apps/web/components/LanguageSelector/flags.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { US } from 'country-flag-icons/string/3x2';
+import { DE } from 'country-flag-icons/string/3x2';
+
+export interface FlagImports {
+  [key: string]: string;
+}
+export const flagImports: FlagImports = {
+  en: US as string,
+  de: DE as string,
+};
```
Comment (FabianGerke) [inline apps/web/components/LanguageSelector/flags.ts:10]: Cant we import these flags in the nuxt.config.ts?
e.g.:

`[
      '@nuxtjs/i18n',
      {
        locales: [
          {
            code: 'en',
            file: 'en.json',
            flag: abcdef,
          },
          {
            code: 'de',
            file: 'de.json',
            flag: abcdef,
          },
        ],
        lazy: true,
        langDir: 'lang',
        defaultLocale: 'en',
      },
    ],`

**Review by FabianGerke — CHANGES_REQUESTED** [review]

File: apps/web/components/LanguageSelector/flags.ts (hunk 5 above)
Comment (rbedeoan-plenty) [reply apps/web/components/LanguageSelector/flags.ts:10]: tried that, vite/nuxt needs specific import from a file ,when dinamically importing

File: apps/web/lang/de.json (hunk 1 above)
Comment (rbedeoan-plenty) [reply apps/web/lang/de.json]: if you search "languageSelector" in project this word was not used anywhere, so i changed it for my usecase to match provided mockups

File: apps/web/lang/en.json (hunk 2 above)
Comment (rbedeoan-plenty) [reply apps/web/lang/en.json]: if you search "languageSelector" in project this word is not use, so i changed it to match provided mockups

File: apps/web/components/LanguageSelector/LanguageSelector.vue (hunk 4 above)
Comment (rbedeoan-plenty) [reply apps/web/components/LanguageSelector/LanguageSelector.vue]: 1.the used classes are tailwind classes
2.if i attach the long stream of classes to the component it seems bloated to me, do we have a code practice to never use .classname ?
I'm open to talk about reduction of tailwind classes and obtaining same result with less tailwind classes..

File: apps/web/lang/en.json (hunk 3 above)
Comment (rbedeoan-plenty) [reply apps/web/lang/en.json]: missed that, typo, fixed it

File: apps/web/lang/de.json (hunk 1 above)
Comment (FabianGerke) [reply apps/web/lang/de.json]: It isn't unused, or why didn't you have to use the translation in your PR?
Used in the default.vue, and I think the translations before were better.

Your translations (english/german) both of them are not correct. There is are no words like en: "languageSelector" and de: "LanguageSelector".

File: apps/web/lang/en.json (hunk 2 above)
Comment (FabianGerke) [reply apps/web/lang/en.json]: Which mockup?

File: apps/web/components/LanguageSelector/LanguageSelector.vue (hunk 4 above)
Comment (FabianGerke) [reply apps/web/components/LanguageSelector/LanguageSelector.vue]:
```
[
      {'all-classes-desktop': isDesktop},
      {'all-classes-mobile': !isDesktop},
]
````

I would say do it that way and remove all the stuff from <style>

**Review by FabianGerke — CHANGES_REQUESTED** [review]
Also there is a bug:
If you set your browser width to like 1000px, than the desktop and mobile language select does not shown.

File: apps/web/lang/en.json (hunk 2 above)
Comment (rbedeoan-plenty) [reply apps/web/lang/en.json]: Screenshot 2023-11-17 at 11 53 24

Comment (rbedeoan-plenty) [reply apps/web/lang/en.json]: you'r right , no mockups for this, it is used only in aria label;
Tim said i should use in aria-label like this="Click to open ...."

File: apps/web/components/LanguageSelector/LanguageSelector.vue (hunk 4 above)
Comment (rbedeoan-plenty) [reply apps/web/components/LanguageSelector/LanguageSelector.vue]: need some context

**Comment by rbedeoan-plenty** [comment]
> Also there is a bug: If you set your browser width to like 1000px, than the desktop and mobile language select does not shown.

found it , let me analyse why

File: apps/web/components/LanguageSelector/LanguageSelector.vue [hunk 6]
```diff
@@ -1,36 +1,50 @@
...
-  <div class="w-full bg-white flex items-center py-10 justify-center flex-col shadow-xl md:hidden">
+  <div
+    data-testid="languageSelectList"
+    :class="[
+      {
+        'flex w-full bg-white items-center py-10 justify-center top-[10%]': isDesktop,
+      },
+      { 'w-full bg-white flex items-center py-10 justify-center flex-col': !isDesktop },
+    ]"
+  >
```
Comment (FabianGerke) [inline apps/web/components/LanguageSelector/LanguageSelector.vue]:
```suggestion
<div
    data-testid="languageSelectList"
    class="w-full bg-white items-center py-10 justify-center flex"
    :class="{ 'top-[10%]': isDesktop, 'flex-col': !isDesktop }"
>
```

File: apps/web/components/LanguageSelector/LanguageSelector.vue [hunk 7]
```diff
@@ -1,36 +1,50 @@
...
+        {{ $t(`lang.${locale}`) }}
+      </div>
+    </SfButton>
   </div>
 </template>
 <script setup lang="ts">
 import { SfButton } from '@storefront-ui/vue';
-
+import { flagImports } from './flags';
+const { isDesktop } = useBreakpoints();
```
Comment (FabianGerke) [inline apps/web/components/LanguageSelector/LanguageSelector.vue]:
```suggestion
import { flagImports } from './flags';

const { isDesktop } = useBreakpoints();
```

**Review by FabianGerke — CHANGES_REQUESTED** [review]
Last small changes, than I can approve it

**Review by FabianGerke — APPROVED** [review]

**Pull request merged**